numpy/_core/numeric.py

```diff
@@ -2531,13 +2530,13 @@ def array_equiv(a1, a2):
     except Exception:
         return False
 
-    return bool((a1 == a2).all())
+    return builtins.bool((a1 == a2).all())
 
 
 inf = PINF
 nan = NAN
-False_ = bool_(False)
-True_ = bool_(True)
+False_ = nt.bool(False)
+True_ = nt.bool(True)
```

Comment:
Using `np.bool` would be clearer; this relies on a weird replacement thing in `numerictypes.py` - it does a `from builtins import bool` and then overwrites that with:
```python
# Now add the types we've determined to this module
for key in allTypes:
    globals()[key] = allTypes[key]
    __all__.append(key)
```

Reply:
I noticed it, but with `np.bool` importing NumPy fails with:

```python
AttributeError: partially initialized module 'numpy' has no attribute 'bool' (most likely due to a circular import)
```
as `numpy.core` module is imported eagerly (`__init__.py` imports `core/numeric.py` which imports `__init__.py`).

I would leave `nt.bool`, or import it locally with a different name? WDYT?

Reply:
Ah okay. In that case, maybe leave it as is, but ensure that there's a test that `_core.numerictypes.bool is np.bool` to guard against a future issue there?

Reply:
Sure! I added a test for that.